### ISSUE TITLE
We need to find a way to give the wiki the unhashed password.

### ISSUE THREAD

**Issue by Radical-from-Scratch** [Newawe/newawe#25]
The wiki does not use sha1() like the forums and (AFAIK) main site do. It does something entirely different. What should we do? No, this issue does not belong in the wiki repo.

**Comment by bjskistad** [comment]
The wiki needs to be connected the db with all the pwords im guessing.

**Comment by Radical-from-Scratch** [comment]
Yeah, I was planning to have the wiki use the same database as the main site.

**Comment by bjskistad** [comment]
Or should we have separate users like the scratch wiki.

**Comment by herohamp** [comment]
Separate like scratch

**Comment by Radical-from-Scratch** [comment]
Well, just because not everyone gets a wiki account doesn't mean they are seperate accounts. It's still llinked  to your Scratch account.

**Comment by Jonathan50** [comment]
Seperate accounts. Just register for the wiki if you need an account.

**Comment by Radical-from-Scratch** [comment]
No, there's a reason Scratch didn't do that and it's a very good one.

**Comment by Jonathan50** [comment]
...Scratch *DID* do that. In Scratch then you can't just log on to the wiki with your Scratch account. You need a wiki account as well.

**Comment by bjskistad** [comment]
I agree with @Jonathan50 Scratch's wiki system is good

**Comment by robinp7720** [comment]
Just use the same auth procedure for the wiki? Can't you just use the same
hash?

**Comment by Radical-from-Scratch** [comment]
I KNOW. THAT'S NOT WHAT I MEAN. What I *actually* meant was that, as I ALREADY STATED, " just because not everyone gets a wiki account doesn't mean they are seperate accounts." I already wanted to do it that way. That's why I COPIED Scratch's system.

**Comment by Radical-from-Scratch** [comment]
@robinp7720 Nope, MediaWiki hashes passwords totally differently than the main site

**Comment by herohamp** [comment]
Make a custom wiki?

**Comment by Radical-from-Scratch** [comment]
God, why does everyone want to write the software themself? We shouldn't write wiki software ourself just because of this one little problem.

**Comment by bjskistad** [comment]
I think all we have to do are a few mods.

**Comment by herohamp** [comment]
Find Radical. Lets just edit MediaWiki

**Comment by Radical-from-Scratch** [comment]
I'm thinking, we'll just do this:

Have them input their main site password
Hold on to their password
check it by sha1() it and check the main site database
Hash the pasword wiki-style.
Done.

**Comment by herohamp** [comment]
That sounds good. But wait for at least 2 more people to say it is good.

**Comment by robinp7720** [comment]
@Radical-from-Scratch, its easy to change the way mediaWiki does authentication.

And if we go your way, I'll have all your passwords within a day, sha1 is childs play for the modern computer.

**Comment by bjskistad** [comment]
lol

**Comment by herohamp** [comment]
... Im scared

**Comment by herohamp** [comment]
XD

**Comment by Radical-from-Scratch** [comment]
1) Do you mean the way it hashes passwords? Oh no it is not.

2) What I meant by sha1() is "whatever the main site uses". And since you work on the main site and I do not, that is your problem/fault.

**Comment by bjskistad** [comment]
Guys, let's stop arguing. We also need to fix the CSS issue.

**Comment by herohamp** [comment]
Agreed!

**Comment by robinp7720** [comment]
@herohamp You should be. Many systems use completaly outdated hashing algorithems. Windows 7 for example, uses md4. Which is even less resource intensive to calculate.

@Radical-from-Scratch If you're not supposed to me making this, could you stop giving advice when you clearly don't know what your talking out.

**Comment by Radical-from-Scratch** [comment]
Why does everyone keep bringing the CSS issue to different  places?!?!

**Comment by herohamp** [comment]
Because it is broken and nothing can be done utill fixed

**Comment by bjskistad** [comment]
lol accidentally assigned as-com.

**Comment by Radical-from-Scratch** [comment]
"Because it is broken" 

Um, yeah.

"nothing can be done utill fixed"

Utill? Anyway, while I agree it's pretty major, that's not really true. It's just css.

**Comment by herohamp** [comment]
Ok. Let me change that. The site will look Ugly AF until it gets fixed

**Comment by Jonathan50** [comment]
This isn't a bug.

**Comment by bjskistad** [comment]
ya...

**Comment by as-com** [comment]
It's probably better to use some sort of single sign on system rather than messing with password hashes.

**Comment by Radical-from-Scratch** [comment]
You mean like O-Auth? Nah, because we want a closed account system.

**Comment by ghost** [comment]
Also, we're moving back to PHP.
